R/data_update.R

```diff
@@ -8,8 +8,15 @@ library(testthat)
 source("R/utils.R")
 
 # Create lookup ===============================================================
-mp_lookup <- dfeR::fetch_pcons(2024, "England") |>
-  dplyr::mutate(pcon_name_lower = tolower(pcon_name)) |>
+mp_lookup <- dfeR::fetch_pcons(2024, "All") |>
+  # Adding a country column to the lookup as it contains multiple countries
+  dplyr::mutate(country = case_when(
+    startsWith(pcon_code, "E") ~ "England",
+    startsWith(pcon_code, "N") ~ "Northern Ireland",
+    startsWith(pcon_code, "S") ~ "Scotland",
+    startsWith(pcon_code, "W") ~ "Wales"),
+    # setting case to lower case as case sensitivity is becoming an issue
+    pcon_name_lower = tolower(pcon_name)) |>
```

Comment:
Given we're adding the names in, I think we might as well as the country codes too, can do that with one extra join from dfeR like this, call the first column `country_name` then join to the countries lookup. You'll need to update the QA to account for this too

```
  # Adding a country column to the lookup as it contains multiple countries
  dplyr::mutate(
    country_name = case_when(
      startsWith(pcon_code, "E") ~ "England",
      startsWith(pcon_code, "N") ~ "Northern Ireland",
      startsWith(pcon_code, "S") ~ "Scotland",
      startsWith(pcon_code, "W") ~ "Wales"
    ),
    # setting case to lower case as case sensitivity is becoming an issue
    pcon_name_lower = tolower(pcon_name)
  ) |>
  dplyr::left_join(dfeR::countries, by = "country_name") |>
```